docs/concepts/controller (#21495)
kubernetes/website: content/it/docs/reference/glossary/controller.md

---
title: Controller
id: controller
date: 2018-04-12
full_link: /docs/concepts/architecture/controller/
short_description: >

>   Un circuito di controllo che osserva lo stato condiviso del cluster attraverso l'API server and apporta le modifiche necessarie per portate lo stato corrente verso lo stato desiderato.

Reviewer: WDYT about
```suggestion
  Un software che implementa un control loop che osserva lo stato condiviso del cluster attraverso l'API server and apporta le modifiche necessarie per portate lo stato corrente verso lo stato desiderato.
```

Author: partially changed due to previous translation agreement

aka: 
tags:
- architecture
- fundamental
---

> In Kubernetes, i _controllers_ sono circuiti di controllo che osservano lo stato del {{< glossary_tooltip term_id="cluster" text="cluster">}}, e apportano o richiedono modifiche quando necessario. Ogni _controller_ prova a portare lo stato corrente del cluster verso lo stato desiderato.

Reviewer: ```suggestion
In Kubernetes, i _controllers_ sono control loop che osservano lo stato del {{< glossary_tooltip term_id="cluster" text="cluster">}}, e apportano o richiedono modifiche quando necessario. Ogni _controller_ prova a portare lo stato corrente del cluster verso lo stato desiderato.
```

Author: not changed due to previous translation agreement

> In Kubernetes, i _controllers_ sono circuiti di controllo che osservano lo stato del {{< glossary_tooltip term_id="cluster" text="cluster">}}, e apportano o richiedono modifiche quando necessario. Ogni _controller_ prova a portare lo stato corrente del cluster verso lo stato desiderato.

Reviewer 2: s/controllers/controller/

<!--more-->

> I _controllers_ osservano lo stato condiviso del cluster attraverso il {{< glossary_tooltip text="apiserver" term_id="kube-apiserver" >}} (che è parte del {{< glossary_tooltip term_id="control-plane" >}}).

Reviewer 2: s/controllers/controller/

> Alcuni _controllers_ vengono eseguiti all'interno del _control plane_, fornendo circuiti di controllo che sono parte dell'operatività base di Kubernetes. Ad esempio: il _deployment_ _controller_, il _daemonset_ _controller_, il _namespace_ _controller_, ed il _persistent volume_

Reviewer: ```suggestion
Alcuni _controllers_ vengono eseguiti all'interno del _control plane_, e forninsconoo dei control loop che sono parte dell'operatività base di Kubernetes. Ad esempio: il _deployment_ _controller_, il _daemonset_ _controller_, il _namespace_ _controller_, ed il _persistent volume_
```

> Alcuni _controllers_ vengono eseguiti all'interno del _piano di controllo_ (_control plane_), e forniscono circuiti di controllo che sono parte dell'operatività base di Kubernetes. Ad esempio: il _deployment_ _controller_, il _daemonset_ _controller_, il _namespace_ _controller_, ed il _persistent volume_

Reviewer 2: s/controllers/controller/

If I remember correctly, it was decided that the control plane should not be translated in Italian, since is the a specific name of an item in Kubernetes

Author: according to last @sftim comment, we decided to translate it and add in parentheses the english word/expression for the first occurrence
